### PR TITLE
Bagstore ingest

### PR DESCRIPTION
**Pull request by rvanheest**
fixes EASY-1127 (bagstore ingest with version-1-deposit)

#### When applied it will
* add a `storage-type` parameter that can either be `dark-archive` (the traditional archive) or `bag-store` (the new archival storage)
* add parameters for the bag-store
* group parameters in `application.properties` such that it is more human-readable
* start a new series of example deposits (will be extended as we move to a *version-n-deposit*)
* change the `Settings` to allow the choice between `dark-archive` and `bag-store`
* add BagStore deposit to `MendeleyExecution`
* create new `TLAExecution`
* redirect TLA deposits to the new `TLAExecution` (using new depositorId `tlatest` in https://github.com/DANS-KNAW/easy-dtap/pull/53)

#### Where should the reviewer @DANS-KNAW/easy start?
every change in this PR

#### How should this be manually tested?
**STILL ON MY TODO LIST**

#### related pull requests on github
repo                                | PR
-----------------------  | -----------------
easy-dtap                       | [PR#53](https://github.com/DANS-KNAW/easy-dtap/pull/53)
easy-ingest-dispatcher | [PR#40](https://github.com/DANS-KNAW/easy-ingest-dispatcher/pull/40)

### CODE REVIEW

**Comment by janvanmansum** [comment]
@rvanheest : is this a draft PR? 

I have tested it, and it doesn't work yet. It looks like you are trying to archive the whole deposit directory as a bag.

**Comment by rvanheest** [comment]
Sorry, forgot to hit commit and push on some later changes...
Please try again. Thanks for the reminder!

**Comment by janvanmansum** [comment]
Okay, so this one *should* be ready for merge then?

File: src/main/scala/nl/knaw/dans/easy/ingest_flow/MendeleyExecution.scala [hunk 1]
```diff
@@ -97,16 +99,41 @@ abstract class MendeleyExecution(implicit override val settings: Settings) exten
...
+      case ds: DarkArchiveSettings => archiveBagInDarkArchive(urn)(ds)
+        .flatMap {
+          case STATE_SUBMITTED => Success(STATE_SUBMITTED)
+          case state => Failure(new IllegalStateException(s"Incorrect state; should be $STATE_SUBMITTED, but was $state"))
+        }
+    }
+  }
+
+  def archiveBagInBagStore(storageSettings: BagStoreSettings): Try[String] = {
+    log.info("Sending bag to BachStore")
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/ingest_flow/MendeleyExecution.scala]: J S Bach or J C Bach? ;)

Comment (rvanheest) [inline src/main/scala/nl/knaw/dans/easy/ingest_flow/MendeleyExecution.scala]: Nope, it's [Wilhelm Friedemann](https://nl.wikipedia.org/wiki/Wilhelm_Friedemann_Bach) 😉 
Gonna fix that!

**Comment by janvanmansum** [comment]
Created and merged a new PR based on this one, solving merge conflicts.

**Pull request closed without merging**